### DISCUSSION TITLE
Latest PCB design

### DISCUSSION THREAD

**Discussion by revk** [revk/xESP32-Faikout#374] [Announcements]
I am constantly improving code and boards.

These are the latest board design, which I expect to be in Amazon maybe October 2024. I'm impressed at KiCad rendering.

![Top](https://github.com/user-attachments/assets/114f812d-eb65-4a1d-97af-c7b2599ce3f6)
![Bottom](https://github.com/user-attachments/assets/74f6d28e-5a4c-4748-92bb-17f43e07be32)

The main changes are.

- Two pads on the back (GPIO 21), the plan is to make a *factory reset* procedure by shorting these.
- New tab that has TC2030 USB programming pads, rather than through hole pads behind the processor.
- Even clearer pin numbering on the S21 connector.

The new tab should be possible to snap off, samples here this week to test that, but should not be necessary because it simply sits behind the connector and hence does not take any real extra space. But it makes programming easier for me, uses a standard lead, and should avoid shorts on back of ESP32 which meant some failed boards in production runs.

Comments welcome.

P.S. Test boards arrived, work nicely, don't get in way of wiring, but also can be snapped off.

![324022](https://github.com/revk/ESP32-Faikin/assets/996983/31d59dd2-8eea-4b57-b0a8-b0fe8be93600)

Update: 2024-08-21

Next batch at Amazon are purple, but the same.

![324342](https://github.com/user-attachments/assets/ed9f77d0-8c41-4964-9398-2bb40f4f62ce)

I am working on the next batch after that (maybe November), which has an additional resistor and diode. This will work on some models that have no pull up on Rx to Faikin, and allow loopback test on Faikin to work again.

**Comment by jmtatsch** [comment]
Seems like amazon ships it directly to Germany even without shipping fee for prime customers.
I am just not sure whether I will keep these Daikin units as they never hold a comfortable temperature. 
Does an external BLE temperature sensor improve things for you?

**Reply by revk** [reply]
![image](https://github.com/revk/ESP32-Faikin/assets/996983/eb62b684-8f86-4b1d-8450-d97f6b9c2e7e)

Indeed, keeps me within a fraction of a degree.

**Comment by Gnec220** [comment]
Hi revk,
i you mentioned in the beta notes that you could do direct sales. Is it possible to ship 2 pcs of the latest version to Scotland ? If so how can i contact you ?

Thank you.

**Reply by revk** [reply]
I have a small batch coming with Rx pull up to 3.3V, should be here next week. Email sales@aa.net.uk FOA Adrian, and I can get sorted when they come in.

**Reply by TheBestMoshe** [reply]
> I have a small batch coming with Rx pull up to 3.3V, should be here next week. Email [sales@aa.net.uk](mailto:sales@aa.net.uk) FOA Adrian, and I can get sorted when they come in.


Any chance you would ship to the US? Or am I better off ordering it through [JLCPCB](https://jlcpcb.com/) (I need 4 of them).

**Reply by revk** [reply]
Maybe, I'd need to check postage costs.

**Comment by vipperhell** [comment]
Hi there.

First of all, thanks for the project. Here in Brazil, no Daikins can be controlled without the Daikin Cloud.

I'm trying to use your kindly shared project, sending it to a PCBA production.

I got stuck on two things.
On BOM list, there were two questions from the production:
1- Component J3, checking on the scheme, is it really just the copper pads? or is there any component indeed?
2- Component L2, there is no part number. I was checking the Mouser website, and there are some of those inductors. Most of them change only the Max Amp. Could you please provide the part number you've used? Or give me one available on Mouser website?

I've tried also to remove your registered marks, but If I try to export only the silk masks, when I see on the Gerber inspector, it shows the masks are offset from the rest of the project. I'm using KiKad, but I'm not very familiar with it.

Last thing is, when I try to render it, I can't see some components, it seems like it's not on the Kikad Library. Are some of components drawings yours? did you share it somewhere?

Sorry for so many questions.

Many Thanks

**Comment by RaW79** [comment]
Is my understandaing correct that due to new layout and according notes it´s now becoming a must to have a TC2030 for reprogramming the module? To be honest I fully understand your comfort in making it mostly efficent for you in mass production but I don´t like hardware beeing so dependent to each other. A very simple USB to Serial converter is a very usual piece of equipment that everybody has in his lab and is available for only 1$. TC2030 is very special and listed for ~40€ (Mor expensive than the whole faikin module). Can not really understand why the 4 Pin design is deleted. Or can you give an information how a reprogramming is still possible without TC2030? Thx. Otherwise it´s more economic to buy a new faikin in case of needed reprogramming instead of buying such a cable -> waste by design!

**Reply by revk** [reply]
There are two reasons.

1. The TC2030 is, indeed, a lot easier to use than the home made cables.
2. The pads behind the CPU resulted in shorts to the back of the CPU and that meant a notable failure rate on new boards. It was a while before I realised how much of a problem this was.

So the old 4 pin header would have to fit somewhere else, making it bigger.

**Reply by RaW79** [reply]
So as long as you are able to operate with your solder iron, this is not a big thing. Just tested and figured out that the TC2030 incl. this 40€ cable is just a simple connection. So e.g. a USB-breakout-board soldered to the pads works fine and that´s it. No need for buying such cable for just one faikin to rescue.

**Reply by revk** [reply]
Yes, that works as well, it is just simple GND/PWR at each and and D-/D+ in the middle.

**Reply by TheBestMoshe** [reply]
@RaW79 Did you end up doing it with your own usb breakout board? Do you have any pictures/schematics you followed?

**Reply by RaW79** [reply]
![PXL_20240908_160503088](https://github.com/user-attachments/assets/6f833d53-a40d-465c-be5c-3c0486633761)

**Comment by vipperhell** [comment]
Hello... 

Was my question above so silly? or did I say something wrong? I'm really new to all this. I'm just trying to have my four Daikin working well in HA.. 
Unfortunately buying it from Amazon is quite expensive because of the redirect service and taxes in Brazil, something I won't face if I buy from any PCB maker because our customs will understand that is not a finished product.

So if you could give me some help, I would appreciate it very much.

Or, If I post this in the wrong place, sorry, and let me know how to move to the correct place.

Thanks.

**Reply by revk** [reply]
Missed that question I think...

- J3 is just copper pads, to work with a TC2030 lead.
- L2 is matched by description at JLCPCB so not sure of expect part, but one of right size and inductance.
- Not sure of your KiCAD issues, sorry.
- Lots of components are in my library, but it should allow you to load them anyway.

**Comment by partytimeexcellent** [comment]
Any downsides to ordering the "older" PCB from Tindie as a USA buyer? I would prefer to purchase the latest, direct from you, but I see you're not offering that, unfortunately.

**Reply by revk** [reply]
Amazon have definitely sold to US on occasion.

They should be fine.

**Comment by DatafoxChris** [comment]
![Daikin_WiFi](https://github.com/user-attachments/assets/f93fa7e3-56b0-4e1e-a1ac-cce4d19b910e)

Anyone know if Faikin module can work when replacing this new Daikin WiFi module ? The connector has 'S90' and 'PCML1652' printed next to it.
If possible, please help me to find the correct converter cable ?

**Reply by RaW79** [reply]
Usually you connect to the connector of the motherboard. This one looks similar to the small PH2.0mm connector but having a look to the motherbord connector should help to say. Which Daikin aircon model?

Seems to be this one: https://mega.nz/folder/aBlETAwZ#UwIzLawwljJ0Xh9-jTLTZw

**Reply by DatafoxChris** [reply]
![Daikin_Doos](https://github.com/user-attachments/assets/d844b950-d5aa-43ce-bac7-769c2884992c)

![Daikin_Moederbord](https://github.com/user-attachments/assets/b7c51505-d7b3-4708-bb9c-609bc7cf5d9b)

The wires from the wifi module seem to be connected to a larger motherboard connector which also has another thing which i cannot see connected to it. So i would like to use the current plug from the daikin wifi module to connect the Faikin board to.

**Reply by RaW79** [reply]
You can use the S403 (10 pole) for the faikin. Works but be careful! One Pin has 230V

See: https://github.com/revk/ESP32-Faikin/issues/208#issuecomment-2173356231

**Reply by DatafoxChris** [reply]
Seems to be axactly my situation. Am i correct in concluding from the pictures in comment #208 that both modules can work simultaneous ?

**Reply by RaW79** [reply]
Yes work in parallel is fine.

**Reply by DatafoxChris** [reply]
![Faikin_Cable](https://github.com/user-attachments/assets/dfe4477a-74d1-4222-a4d0-86ee05b6f32a)

You mean like this ;)

**Reply by revk** [reply]
Jumper wire also works

**Reply by DatafoxChris** [reply]
I didn't notice that red is on the first pin on the motherboard, so i 'tweaked' my JST connector with cutters

**Reply by RaW79** [reply]
Yes, according your personal taste. I like more fix/stable solutions but all roads leads to Rome:
![grafik](https://github.com/user-attachments/assets/c8c9fa54-e680-43fe-9228-a27cb995a2a4)

**Reply by DatafoxChris** [reply]
Wow, it actually works :)

**Comment by RaW79** [comment]
May I kindly ask if the additional resistor and diode are included by the 4th November design release? Is there sth. else changed? Was interested about changes and could see the comment "fine tune". Thx

**Reply by revk** [reply]
You can work out why when changes were made, but the change was to allow a clean pull up of Tx line to 5V, but also handle possible 5V being connected to GND somehow. In practice we find Tx and Rx have internal pull ups anyway, hence open drain driving of Tx, but that some systems need a pull up, hence resistor to 5V. The addition of the diode is just extra protection for 5V being GND, if that makes sense.
